homeassistant/components/rfxtrx.py

```diff
@@ -261,25 +262,28 @@ def get_pt2262_device(device_id):
 def find_possible_pt2262_device(device_id):
     """Look for the device which id matches the given device_id parameter."""
     for dev_id, device in RFX_DEVICES.items():
-        if len(dev_id) == len(device_id):
-            size = None
-            for i in range(0, len(dev_id)):
-                if dev_id[i] != device_id[i]:
-                    break
-                size = i
-
-            if size is not None:
-                size = len(dev_id) - size - 1
-                _LOGGER.info("rfxtrx: found possible device %s for %s "
-                             "with the following configuration:\n"
-                             "data_bits=%d\n"
-                             "command_on=0x%s\n"
-                             "command_off=0x%s\n",
-                             device_id,
-                             dev_id,
-                             size * 4,
-                             dev_id[-size:], device_id[-size:])
-                return device
+        try:
+            if device.is_lighting4 and len(dev_id) == len(device_id):
```

Comment:
If you do `isinstance(RfxtrxBinarySensor) and  device.is_lighting4  ...` you can avoid the the try-except ?

Reply:
I've never been able to use "isinstance(RfxtrxBinarySensor)" because of cyclic imports. Using "pass" does not work either. I'd be glad if someone can help me solve this!

Reply:
I see.
But can use `hasattr(device, 'is_lighting4')` ?